base/session_local.cpp

```diff
@@ -32,16 +34,19 @@ std::vector<std::function<void()>>& SessionLocal::get_finalizers() {
     return finalizers;
 }
 
-std::vector<std::function<void()>>& SessionLocal::get_thread_finalizers() {
-    static thread_local std::vector<std::function<void()>> thread_finalizers;
+std::vector<std::pair<int, std::function<void()>>>& SessionLocal::get_thread_finalizers() {
+    static thread_local std::vector<std::pair<int, std::function<void()>>> thread_finalizers;
     return thread_finalizers;
 }
 
 void SessionLocal::register_initializer(std::function<void()> init) { get_initializers().push_back(init); }
 
+
 void SessionLocal::register_finalizer(std::function<void()> fina) { get_finalizers().push_back(fina); }
 
-void SessionLocal::register_thread_finalizer(std::function<void()> fina) { get_thread_finalizers().push_back(fina); }
+void SessionLocal::register_thread_finalizer(int priority, std::function<void()> fina) {
```

Comment:
So please give the description about the priority. If a new one watches this function, he has to find where the priority is used to understand.